configure.py

```diff
@@ -485,8 +485,9 @@
 parser.add_option('--code-cache-path',
     action='store',
     dest='code_cache_path',
-    help='Use a file generated by tools/generate_code_cache.js to compile the'
-         ' code cache for builtin modules into the binary')
+    help='optparse.SUPPRESS_HELP')
```

Comment:
This can be removed also, I believe.

Reply:
I'd rather not make this semver-major so we can get this into 12

Reply:
Why would we make this semver-major? This is just a configure-time switch.

Reply:
Build time switches are part of the support "contact" in the context of downstream channels. i.e. _if_ someone was using it, removing will break their build.

Reply:
But how would they use this switch? The original script to produce file does not exist now so if the build is using it, it's already broken, and they have to make adjustment anyway.

Reply:
Well it doesn't do anything, and `make with-code-cache` does nothing so running:
```console
$ ./configure
configure completed successfully
$ make with-code-cache
"'with-code-cache' target is a noop
$ ./configure --code-cache-path=dlfguhzdlkfjghd,jbnvsghsaiuthg
configure completed successfully
$ make
...
```
Will just work 🤷‍♂️

Reply:
> $ ./configure --code-cache-path=dlfguhzdlkfjghd,jbnvsghsaiuthg
> configure completed successfully

But where would `dlfguhzdlkfjghd,jbnvsghsaiuthg` come from? They would have to change their command to generate it now.

Reply:
It doesn't really matter, the value is never used by us. If before they had `out/Release/obj/gen/node_code_cache.cc` hardcoded or any other value for `$CODE_CACHE_FILE` in they scripts the build should not break.

It will only break is they actually check that the file is there
If they generate it themselves somehow else, they will get UB.

Reply:
I'm not super attached to this, I just don't want create unexpected breaks. We could instead do some communicating with the delivery channels...

Reply:
The thing is the only valid generator has been tools/generate_code_cache.js, if they used the script in their build, their build breaks since the script no longer exists; If they use something else, it also breaks because the generated format is now changed with the previous refactoring patch. So I don’t see the point keeping this switch if the build relying on this breaks and has to be fixed either way.

> We could instead do some communicating with the delivery channels...

I agree, but before that happens, I don’t think it’s a good idea to be too careful and create unnecessary maintaince burden. If this turns out to be an issue, someone will open an issue, but from my experience managing a fork in my previous job and talking to other people doing similar work, people always prefer making adjustments for upstream changes instead of having upstream adapt to them because the upstream tend to guess wrong - a dummy target is not really a useful to help, if one previously uses this, you’d want to know and remove dead code now instead of keeping some invalid commands around in your script.

Reply:
To be clear, I am not requesting to change this noop in this patch - it’s perfectly fine to leave it as noop if of makes you feel comfortable. I am just not convinced that changes like this should be semver major or this is part of our support contract - if there was such a contract, we have already broken this too many times to continue supporting it.